### ISSUE TITLE
weighted Mean Square Error (wMSE) loss

### ISSUE THREAD

**Issue by HajarMazaheri** [jiequancui/DKL#2]
Hi. Can we use `nn.MSELoss()` to calculate (`wMSE`)?
Thank you for guiding me

**Comment by jiequancui** [comment]
Hi,

Thanks for your interest in our work.

I think it is equivalent to implement it directly by torch.nn.MSELoss.


Best,
Jiequan

**Comment by HajarMazaheri** [comment]
Thank you Jiequan, @jiequancui 

I work in Speech Recognition with knowledge distillation technique. I have already used the `KL+CE` loss function and MSE or cosine_similarity loss for intermediate layers. I am interested in the idea of your article and would like to use the DKL loss. Do you think my implementation method for this function is correct?

`    def wMSE(target_distribution, log_predicted_distribution, labels):
        MSE_loss = nn.MSELoss(reduction="none")
        wMSE_loss= MSE_loss(log_predicted_distribution, target_distribution)

        padding_mask = labels >= 0
        padding_mask = padding_mask.unsqueeze(-1)

        wMSE_loss = wMSE_loss * padding_mask
        wMSE_loss = wMSE_loss.sum() / padding_mask.sum()    
        return wMSE_loss

student_distribution = nn.functional.log_softmax(student_outputs.logits, dim=-1)
teacher_distribution = nn.functional.softmax(teacher_outputs.logits, dim=-1)

wMSE_Loss = wMSE(teacher_distribution, student_distribution, batch["labels"])
ce_loss = student_outputs.loss

DLK_loss = 0.8 * ce_loss + 0.25 *(training_args.MSE_weight *
 wMSE_Loss))  

MSE_weight: Optional[float] = field(default=1.0,metadata={"help": (
   "Weighting assigned to the MSE loss in the KD formulation. MSE loss is "
    "computed between the teacher-student hidden states and attentions.")}"
`

Thank you for guiding me
‌Best,
Hajar
